crates/typst-library/src/text/shaping.rs

```diff
@@ -86,21 +90,24 @@ pub struct Adjustability {
 impl ShapedGlyph {
     /// Whether the glyph is a space.
     pub fn is_space(&self) -> bool {
-        matches!(self.c, ' ' | '\u{00A0}' | '　')
+        is_space(self.c)
     }
 
     /// Whether the glyph is justifiable.
     pub fn is_justifiable(&self) -> bool {
         // GB style is not relevant here.
-        self.is_space()
-            || self.is_cjk_script()
-            || self.is_cjk_left_aligned_punctuation(true)
-            || self.is_cjk_right_aligned_punctuation()
-            || self.is_cjk_center_aligned_punctuation(true)
+        self.is_justifiable
+    }
+
+    /// Updates the justifiability of the glyph.
+    fn update_justifiable(&mut self) {
```

Comment:
Sorry for the late comment after the PR merged. But the `update_justifiable` function is not needed. Though the width of a glyph may change due to some CJK adjustments, whether the glyph is justifiable will never change.

Reply:
Awesome thanks for noticing I'll open a follow up or with that later this afternoon, even more optimization 🎉

Reply:
You're absolutely right, the reason why I added the `update_justifiable` was because I assumed it changes (since it relies on the width) but it does not, I'll open up a follow up PR, Thanks a lot!